src/Plugin/migrate/source/YAML.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\migrate_default_content\Plugin\migrate\source;
+
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Plugin\migrate\source\SourcePluginBase;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * Source for YAML.
+ *
+ * If the YAML file contains non-ASCII characters, make sure it includes a
+ * UTF BOM (Byte Order Marker) so they are interpreted correctly.
+ *
+ * @MigrateSource(
+ *   id = "yaml"
+ * )
+ */
+class YAML extends SourcePluginBase {
+
+  /**
+   * List of available source fields.
+   *
+   * Keys are the field machine names as used in field mappings, values are
+   * descriptions.
+   *
+   * @var array
+   */
+  protected $fields = [];
+
+  /**
+   * List of key fields, as indexes.
+   *
+   * @var array
+   */
+  protected $keys = [];
+
+  /**
+   * The file class to read the file.
+   *
+   * @var string
+   */
+  protected $fileClass = '';
+  
+  /**
+   * The human-readable column headers, keyed by column index in the CSV.
+   *
+   * @var array
+   */
+  protected $columnNames = [];
+
+  /**
+   * The file object that reads the YAML file.
+   *
+   * @var \SplFileObject
+   */
+  protected $file = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $migration);
+
+    // Path is required.
+    if (empty($this->configuration['path'])) {
+      throw new MigrateException('You must declare the "path" to the source YAML file in your source settings.');
+    }
+
+    // Key field(s) are required.
+    if (empty($this->configuration['keys'])) {
+      throw new MigrateException('You must declare "keys" as a unique array of fields in your source settings.');
+    }
+
+//     $this->fileClass = empty($configuration['file_class']) ? 'Drupal\migrate_source_YAML\YAMLFileObject' : $configuration['file_class'];
+  }
+
+  /**
+   * Return a string representing the source file path.
+   *
+   * @return string
+   *   The file path.
+   */
+  public function __toString() {
+    return $this->configuration['path'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function initializeIterator() {
+
+    $parser = new \Symfony\Component\Yaml\Parser();
+    $ret = new \ArrayIterator(
+        $parser->parse(file_get_contents($this->configuration['path']))
+    );
+    
+    if (empty($this->configuration['column_names'])) {
+        foreach (array_keys($ret->offsetGet(0)) as $header) {        
```

Comment:
They are just different coding standards. If you come from symfony it might be a bit painful but we prefer not to mix the coding standars 
The files to feed to phpcs are in the following project
https://www.drupal.org/project/coder
I can assist you further with this if you want.